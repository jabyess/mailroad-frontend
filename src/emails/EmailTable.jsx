import React from 'react'
import PropTypes from 'prop-types'
import EmailTableRow from './EmailTableRow.jsx'

class EmailTable extends React.Component {
	
	render() {
		return (
			<table className="table is-striped email-table">
				<thead className="email-table__header">
					<tr>
						<th>Select</th>
						<th>Title</th>
						<th>Author</th>
						<th>Template</th>
						<th>Category</th>
						<th>Created</th>
						<th>Updated</th>
					</tr>
				</thead>
				<tbody>
					{this.props.emailItems.map((cv) => {
						return (
							<EmailTableRow 
								rowValues={cv}
								key={cv.id}
								checked={this.props.selectedCheckboxes[cv.id]}
								updateSelectedCheckboxes={this.props.updateSelectedCheckboxes}
							/>
						)
					})}
				</tbody>
			</table>
		)
	}
}

EmailTable.propTypes = {
	selectedCheckboxes: PropTypes.object,
	updateSelectedCheckboxes: PropTypes.func,
	emailItems: PropTypes.array
}

export default EmailTable