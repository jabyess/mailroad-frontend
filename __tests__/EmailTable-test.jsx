import React from 'react'
import EmailTable from '../jsx/emails/EmailTable.jsx'
import renderer from 'react-test-renderer'

const emails = [
  {id: 1},
  {id: 2},
  {id: 3},
  {id: 4}
]

const noop = () => {}
const selected = {}

// Test: does an email table get created properly?
it('Creates a table of n emails', () => {
	const component = renderer.create(
    <EmailTable
      emailItems={emails}
      selectedCheckboxes={selected}
      updateSelectedCheckboxes={noop}
    >
    </EmailTable>
  )

	const tree = component.toJSON()
	const selectors = tree.children

	expect(selectors[0].type).toBe('thead')
	expect(selectors[1].type).toBe('tbody')

	const trows = selectors[1].children
	expect(trows.length).toBe(emails.length)

	const first_row = trows[0]
	expect(first_row.type).toBe('tr')

  expect(tree).toMatchSnapshot()
})
