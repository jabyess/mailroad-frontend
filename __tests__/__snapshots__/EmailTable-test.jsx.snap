// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates a table of n emails 1`] = `
<table
  className="table is-striped email-table"
>
  <thead
    className="email-table__header"
  >
    <tr>
      <th>
        Select
      </th>
      <th>
        Title
      </th>
      <th>
        Author
      </th>
      <th>
        Template
      </th>
      <th>
        Category
      </th>
      <th>
        Created
      </th>
      <th>
        Updated
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="email-table__row"
    >
      <td
        className="email-table__row__select"
      >
        <input
          className="checkbox"
          id="checkbox-1"
          onChange={[Function]}
          type="checkbox"
          value={1}
        />
      </td>
      <td
        className="email-table__row__title"
      >
        <a
          onClick={[Function]}
          style={Object {}}
        />
      </td>
      <td
        className="email-table__row__author"
      />
      <td
        className="email-table__row__template"
      />
      <td
        className="email-table__row__category"
      />
      <td
        className="email-table__row__created-date"
      >
        2017-06-20 11:20 AM
      </td>
      <td
        className="email-table__row__updated-date"
      >
        2017-06-20 11:20 AM
      </td>
    </tr>
    <tr
      className="email-table__row"
    >
      <td
        className="email-table__row__select"
      >
        <input
          className="checkbox"
          id="checkbox-2"
          onChange={[Function]}
          type="checkbox"
          value={2}
        />
      </td>
      <td
        className="email-table__row__title"
      >
        <a
          onClick={[Function]}
          style={Object {}}
        />
      </td>
      <td
        className="email-table__row__author"
      />
      <td
        className="email-table__row__template"
      />
      <td
        className="email-table__row__category"
      />
      <td
        className="email-table__row__created-date"
      >
        2017-06-20 11:20 AM
      </td>
      <td
        className="email-table__row__updated-date"
      >
        2017-06-20 11:20 AM
      </td>
    </tr>
    <tr
      className="email-table__row"
    >
      <td
        className="email-table__row__select"
      >
        <input
          className="checkbox"
          id="checkbox-3"
          onChange={[Function]}
          type="checkbox"
          value={3}
        />
      </td>
      <td
        className="email-table__row__title"
      >
        <a
          onClick={[Function]}
          style={Object {}}
        />
      </td>
      <td
        className="email-table__row__author"
      />
      <td
        className="email-table__row__template"
      />
      <td
        className="email-table__row__category"
      />
      <td
        className="email-table__row__created-date"
      >
        2017-06-20 11:20 AM
      </td>
      <td
        className="email-table__row__updated-date"
      >
        2017-06-20 11:20 AM
      </td>
    </tr>
    <tr
      className="email-table__row"
    >
      <td
        className="email-table__row__select"
      >
        <input
          className="checkbox"
          id="checkbox-4"
          onChange={[Function]}
          type="checkbox"
          value={4}
        />
      </td>
      <td
        className="email-table__row__title"
      >
        <a
          onClick={[Function]}
          style={Object {}}
        />
      </td>
      <td
        className="email-table__row__author"
      />
      <td
        className="email-table__row__template"
      />
      <td
        className="email-table__row__category"
      />
      <td
        className="email-table__row__created-date"
      >
        2017-06-20 11:20 AM
      </td>
      <td
        className="email-table__row__updated-date"
      >
        2017-06-20 11:20 AM
      </td>
    </tr>
  </tbody>
</table>
`;
